components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -2169,6 +2207,49 @@ std::string AdsServiceImpl::LoadResourceForId(const std::string& id) {
   return LoadDataResourceAndDecompressIfNeeded(resource_id);
 }
 
+void AdsServiceImpl::ClearScheduledCaptcha() {
+#if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
+  adaptive_captcha_service_->ClearScheduledCaptcha();
+#endif
+}
+
+void AdsServiceImpl::GetScheduledCaptcha(
+    const std::string& payment_id,
+    ads::GetScheduledCaptchaCallback callback) {
+#if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
+  adaptive_captcha_service_->GetScheduledCaptcha(
+      payment_id, base::BindOnce(&AdsServiceImpl::OnGetScheduledCaptcha,
+                                 AsWeakPtr(), std::move(callback)));
+#endif
+}
+
+#if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
+void AdsServiceImpl::OnGetScheduledCaptcha(
```

Comment:
this seems unnecessary, why wouldn't we just pass the callback to `adaptive_captcha_service_->GetScheduledCaptcha`?

Reply:
Right, fixed.